### ISSUE TITLE
ORM specific suggestions

### ISSUE THREAD

**Issue by sivagollapalli** [bbatsov/rubocop#2290]
Thanks for great gem. It helped a lot to follow the conventions and ruby standards. My project uses `mongoid` as a ORM. I have following snippet on project:

```
  Project.all.each do |p|
    # code here
  end
```
`rubocop` gave warning as Use `find_each` instead of `each`. As matter of fact there is no such method on `mongoid` to use. Could we avoid these if particular rails project uses `mongoid` as a ORM?

**Comment by palkan** [comment]
Just turn the Rails/FindEach cop off within your config file:

```yaml
# .rubocop.yml
Rails/FindEach:
  Enabled: false
```

**Comment by sivagollapalli** [comment]
@palkan As of now I have disabled in the same manner. But could we silent these kind of warnings without disabling the cop?

**Comment by palkan** [comment]
What do you mean by "silent this kind of warnings"? What's wrong with disabling?

**Comment by maxjacobson** [comment]
I think the issue is that inexperienced programmers might include RuboCop in their project and follow its advice blindly, and get some bad advice from RuboCop.

In theory, RuboCop could detect whether Mongoid is in use in a project and register offenses only when it is not. That's a bit complicated, but probably doable?

IMO, it could be resolved by explaining in the README/help documentation that if you're going to enable the Rails cops, be mindful that some of them are written with active record in mind.

**Comment by palkan** [comment]
RuboCop doesn't consider project as a whole, only separate files. Thus, it doesn't know that you are using Mongoid.

I think that it possible to extend this cop's message with smth like: "Note: This cop only works with ActiveRecord. If you are using another ORM, you should disable it". @bbatsov

**Comment by bbatsov** [comment]
Another approach would be to add a config option about the ORM, which some cops would take into account.

**Comment by sivagollapalli** [comment]
@bbatsov +1

**Comment by alexdowad** [comment]
> Another approach would be to add a config option about the ORM

I suggest that when `rubocop-rails` is split off into a separate gem, whoever maintains that gem can deal with this issue. If they want to add a new config param, fine. If they don't want to, fine.

**Comment by alexdowad** [comment]
Maintainers, can we close this issue?

**Comment by maxjacobson** [comment]
@bbatsov @alexdowad I had an idea today that I want to share about these rails-specific issues that occur. Rather than splitting out `rails-rubocop`, split out

* `activesupport-rubocop`
* `activerecord-rubocop`
* `mongoid-rubocop` (optionally :smile:)

Doesn't it make sense?

Either way, I'm interested in helping with this process.

**Comment by bbatsov** [comment]
Sounds reasonable. The only small problem is that if you're not assuming the presence of a Rails project you can't really guess which files to inspect with those cops, but I don't think the users would mind dealing with a bit of extra configuration. Or there can be some option to assume you're in a Rails project...
